tensorflow/core/kernels/sparse_to_dense_op_gpu.cu.cc

```diff
@@ -0,0 +1,220 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#if GOOGLE_CUDA
+#define EIGEN_USE_GPU
+
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/common_runtime/gpu/gpu_event_mgr.h"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/framework/types.h"
+#include "tensorflow/core/kernels/sparse_to_dense_op_gpu.h"
+#include "tensorflow/core/platform/stream_executor.h"
+#include "tensorflow/core/util/gpu_kernel_helper.h"
+
+namespace tensorflow {
+
+namespace {
+
+template <typename T, typename Index>
+__global__ void SparseToDenseKernel(
+    const Index* __restrict__ indices, const T* __restrict__ vals,
+    const int nnz, const int num_vals,
+    const Index* __restrict__ dims, const int ndims, T* __restrict__ dense) {
+  GPU_1D_KERNEL_LOOP(thread_idx, nnz) {
+    eigen_assert(ndims >= 1);
+    Index output_idx = indices[thread_idx * ndims + ndims - 1];
```

Comment:
Should we use `int64` for `output_idx`?  Otherwise we risk overflowing when `Index` is `int32`.

Reply:
Done. PTAL.